### ISSUE TITLE
mIoU of SFNet is much lower than 78.9%

### ISSUE THREAD

**Issue by perrying** [donnyyou/torchcv#162]
I trained SFNet with default setting of ```torchcv/scripts/seg/cityscapes/run_sfnet_res18_cityscapes.sh``` except for ```NGPUS``` (I changed ```NGPUS``` from 8 to 4, and ```--train_batch_size``` from 2 to 4).
But I got about 73% of mIoU for single scale inference.
It may be because pretrained model, ```3x3resnet18-imagenet.pth```, is not given.
Although I also trained SFNet (ResNet-101) using pretrained model, ```3x3resnet101-imagenet.pth```, given by this repository, I got about 78% of mIoU for multi-scale inference.

How can I reproduce the paper results?

**Comment by seekFire** [comment]
@perrying  Hi, Have you achieved the result shown in the paper yet?

**Comment by perrying** [comment]
I haven't reproduced it yet.

**Comment by seekFire** [comment]
@perrying Thank you for your reply! I was interested in this work and planned to re-implement it, but now it seems that my plan should be postponed...

**Comment by lxtGH** [comment]
@seekFire @perrying We will release the trained model. Also, we will release the better models in another repo. https://github.com/lxtGH/SFSegNets

**Comment by seekFire** [comment]
@lxtGH Thank you for your response, we are looking forward to it!

**Comment by lxtGH** [comment]
@donnyyou Please give the trained ckpt for reproducing results.

**Comment by donnyyou** [comment]
You could email your training logs to "youansheng@pku.edu.cn", and I will check them for you. @perrying @seekFire

**Comment by donnyyou** [comment]
The logs you email me seems right for res101, but wrong for res18. The pretrained res18 model is missing for initialing the weights. You could download the deepbase res18 pretrained model to fix this.

**Comment by donnyyou** [comment]
@perrying

**Comment by donnyyou** [comment]
https://github.com/CSAILVision/semantic-segmentation-pytorch/blob/master/mit_semseg/models/resnet.py
The pretrained imagenet models could be downloaded here! Note that the resnet version is different from the torchvision version by replacing 7x7 convolution with three 3x3 convolutions.

**Comment by donnyyou** [comment]
Not using deeply supervised losses, that res18-based model uses, might get better results for training the sf models with deeper resnets, such as res50, res101, etc..

**Comment by donnyyou** [comment]
More questions please email me at youansheng@pku.edu.cn.

**Comment by donnyyou** [comment]
You should also change the loss type to fpndsnohemce_loss series for training res18-based sf model. @perrying

**Comment by seekFire** [comment]
@donnyyou So you mean when training the sf model with deeper resnets, we should not use fpndsnohemce_loss series as its loss, isn't it?

**Comment by perrying** [comment]
@donnyyou Thank you for the clarification!

**Comment by donnyyou** [comment]
> @donnyyou So you mean when training the sf model with deeper resnets, we should not use fpndsnohemce_loss series as its loss, isn't it?

You are right, and dsnce_loss might get better results.

**Comment by seekFire** [comment]
@perrying OK, thanks for your answer! BTW, if I select other backbone, such as HRNet, MobileNet and so on, which loss would you recommend?

**Comment by donnyyou** [comment]
deeply supervised losses for smaller models. @seekFire

**Comment by seekFire** [comment]
@donnyyou OK! I see. Thank you very much!
